lib/util/SortableSet.js

```diff
@@ -1,19 +1,32 @@
 "use strict";
-//TODO: Make this a generic type
-//https://github.com/Microsoft/TypeScript/issues/23385
-//https://github.com/Microsoft/TypeScript/issues/23384
+
+/**
+ * A subset of Set that offers sorting functionality
+ * @template T item type in set
+ */
 class SortableSet extends Set {
+	// eslint-disable-next-line valid-jsdoc
+	/**
+	 * Create a new sortable set
+	 * @param {Array<T>=} initialIterable The initial iterable value
+	 * @param {SortFunction=} defaultSort Default sorting function
+	 * @typedef {(a: T, b: T) => number} SortFunction
```

Comment:
Put the typedef in a separate comment.

Actually `=>` is not valid jsdoc. You should use `@callback` or `{function(T, T): number}`